extensions/datastores/kudu/src/main/java/org/locationtech/geowave/datastore/kudu/operations/KuduDeleter.java

```diff
@@ -27,21 +27,86 @@ public KuduDeleter(final KuduOperations operations, final String tableName) {
   @Override
   public void delete(final GeoWaveRow row) {
     try {
+      KuduTable table = operations.getTable(tableName);
+      Schema schema = table.getSchema();
+      List<KuduPredicate> preds = new ArrayList<>();
+      byte[] partitionKey = row.getPartitionKey();
+      short adapterId = row.getAdapterId();
+      byte[] sortKey = row.getSortKey();
+      byte[] dataId = row.getDataId();
+      int numDuplicates = row.getNumberOfDuplicates();
       for (GeoWaveValue value : row.getFieldValues()) {
-        Delete delete = operations.getDelete(tableName);
-        PartialRow partialRow = delete.getRow();
-        operations.addToPartialRow(row, value, partialRow, null);
-        session.apply(delete);
+        preds.add(
```

Comment:
Can you add a comment stating why we need to first scan instead of deleting directly?